### PR TITLE
Support for multiple routes

### PR DESCRIPTION
**Pull request by dantleech**
Integrates: https://github.com/symfony-cmf/RoutingAuto/pull/62

Note that this PR revealed a bug: https://github.com/symfony-cmf/RoutingAutoBundle/issues/171 and I do not currently think it is related to this PR.

### CODE REVIEW

File: Resources/config/auto_route.xml [hunk 1]
```diff
@@ -82,10 +87,9 @@
         </service>
 
         <service id="cmf_routing_auto.metadata.factory"
-            class="%cmf_routing_auto.metadata.factory.class%"
-            factory-service="cmf_routing_auto.metadata.factory.builder"
-            factory-method="getMetadataFactory"
-            />
+            class="%cmf_routing_auto.metadata.factory.class%">
+            <factory service="cmf_routing_auto.metadata.factory.builder" method="getMetadataFactory" />
```
Comment (dantleech) [inline Resources/config/auto_route.xml]: Hmm .. this is 2.6 only, should I revert this..

File: Tests/Resources/app/config/routing_auto.yml [hunk 2]
```diff
@@ -12,7 +12,19 @@ Symfony\Cmf\Bundle\RoutingAutoBundle\Tests\Resources\Document\Post:
...
+            defaults:
+                _type: article_view
+        edit:
+            template: /articles/{article_locale}/{article_title}-edit
+            defaults:
+                _type: article_edit
+        review:
+            template: /articles/{article_locale}/{article_title}-review
+            defaults:
+                _type: article_review
```
Comment (dantleech) [inline Tests/Resources/app/config/routing_auto.yml]: Assert that defaults are stored (currently I don't think they are)

**Comment by dantleech** [comment]
@WouterJ this build is failing because of the new deprecation notices, apparently -- am I missing something?

**Comment by dantleech** [comment]
Failing on lowest dependencies because of deprecations: https://travis-ci.org/symfony-cmf/RoutingAutoBundle/jobs/97946914 /cc @WouterJ

**Comment by WouterJ** [comment]
@dantleech you have to use the new syntax wherever possible. For tests testing the BC layers, add `/** @group legacy */` to the test class/test method.

**Comment by dantleech** [comment]
@WouterJ would that fix the tests? they work in all other environments other than the `--prefer-lowest` one.

**Comment by WouterJ** [comment]
@dantleech other tests are run with weak mode for deprecations (that's not really great, but let's change that in another PR), so deprecation notices won't show up as errors.

The problem is that many of the test fixtures still use strings for uri schema, e.g. https://github.com/symfony-cmf/RoutingAutoBundle/blob/multiple_routes/Tests/Resources/app/config/routing_auto.yml

**Comment by dantleech** [comment]
Yeah I can fix that, but the tests pass in all the other envs. I guess it is an earlier version of the deprecation code that makes it fail with the lowest deps? Will adding the `@group legacy` stop it failing?

**Comment by WouterJ** [comment]
@dantleech tests pass in other envs, as the PHPUnit deprecation helper is set to weak.

We should avoid using deprecated things as much as possible, so please fix the test configuration :)

**Comment by dantleech** [comment]
Fixed, however still getting deprecations for the `factory-service` attributes of the DI container.

```
Symfony\Component\DependencyInjection\Definition::setFactoryService(cmf_routing_auto.metadata.factory.builder) is deprecated since version 2.6 and will be removed in 3.0. Use Definition::setFactory() instead: 5x
    1x in CmfRoutingAutoExtensionTest::testAutoMappingRegistration from Unit\DependencyInjection
    1x in CmfRoutingAutoExtensionTest::testOverrideImplicitPhpcrOdmAdapter from Unit\DependencyInjection
    1x in CmfRoutingAutoExtensionTest::testExplicitAdapter from Unit\DependencyInjection
    1x in CmfRoutingAutoExtensionTest::testLoad from Unit\DependencyInjection
    1x in CmfRoutingAutoExtensionTest::testImplicitPhpcrOdmAdapter from Unit\DependencyInjection

The "factory-service" attribute of service "cmf_routing_auto.metadata.factory" in file "/home/daniel/www/symfony-cmf/RoutingAutoBundle/DependencyInjection/../Resources/co
nfig/auto_route.xml" is deprecated since version 2.6 and will be removed in 3.0. Use the "factory" element instead: 5x
    1x in CmfRoutingAutoExtensionTest::testAutoMappingRegistration from Unit\DependencyInjection
    1x in CmfRoutingAutoExtensionTest::testOverrideImplicitPhpcrOdmAdapter from Unit\DependencyInjection
    1x in CmfRoutingAutoExtensionTest::testExplicitAdapter from Unit\DependencyInjection
    1x in CmfRoutingAutoExtensionTest::testLoad from Unit\DependencyInjection
    1x in CmfRoutingAutoExtensionTest::testImplicitPhpcrOdmAdapter from Unit\DependencyInjection
```

**Comment by dantleech** [comment]
Only one deprecation warning now - about the `translateObject` having to return an object, when using the lowest deps. it is using an older version of the component. Should we boost the requirement for the RoutingAuto component?

**Comment by dantleech** [comment]
Ignore last comment -- all the deprecations are fixed.

File: Resources/config/auto_route.xml [hunk 3]
```diff
@@ -24,7 +25,7 @@
     <services>
         <service
             id="cmf_routing_auto.slugifier"
-            class="Symfony\Cmf\Bundle\CoreBundle\Slugifier\CallbackSlugifier"
+            class="Symfony\Cmf\Api\Slugifier\CallbackSlugifier"
```
Comment (WouterJ) [inline Resources/config/auto_route.xml]: this should also be merged into 1.1, I guess?

Comment (dantleech) [inline Resources/config/auto_route.xml]: I think so.

Comment (dantleech) [inline Resources/config/auto_route.xml]: Created issue https://github.com/symfony-cmf/RoutingAutoBundle/issues/179

**Comment by stof** [comment]
> tests pass in other envs, as the PHPUnit deprecation helper is set to weak.

@WouterJ why is the helper set to weak for non-lowest deps ? This looks weird to me. I generally do the opposite (as lowest deps might end up triggering some deprecations by uisng a version of the deps before they stopped using the deprecated API of the other package)

**Comment by WouterJ** [comment]
@stof I've improved it in the 1.1 branch: https://github.com/symfony-cmf/RoutingAutoBundle/commit/724db6a894459033977e757c6091f75e896d8505 The original config was from some months back: Lower versions didn't have any deprecations and the newer ones were having deprecations caused by third party bundles.

**Pull request merged**